VERSIONING.md

```diff
@@ -0,0 +1,55 @@
+Rocket Chip Versioning Strategy ** All Strawman to be discussed**
+===============================
+
+Version Naming
+--------------
+
+* Rocket Chip uses a Calendar Versioning release strategy (https://calver.org/).
+* The format is YYYY.MM.DD.MINOR
+* MINOR changes include:
+  * API Additions
+  * Bug Fixes (which can result in changes in the emitted hardware considered bugs)
+* MINOR changes do NOT include:
+  * Changes in top-level I/Os for < level of hierarchy >
+  * Backwards incompatible changes in the API (see below)
+
+API
+---
+
+The Rocket-Chip API is: 
+  * All public Scala methods and classes
+  * All public Scala methods and classes of included Scala libraries (hardfloat, CDE, etc)
+  * The command line used to invoke the generator
+  * All Verilog interfaces (I/Os and parameterization) for verilog files
+  * C code...?
+  * The version(s) of Chisel, FIRRTL, rocket-tools pointed to by rocket-chip.
```

Comment:
Is the intention here to have an `=` or a `>=` relationship with the Chisel/FIRRTL/etc. versions?

Reply:
i think my intention is if you bump the major version of CHisel/FIRRTL, that would count as a major (not MINOR) change. But bumping the Chisel/FIRRTL version to one that is also backwards compatible would be considered a MINOR change.

Does that answer your question? How can I make it more clear?

Reply:
Ah, I see. That makes sense. I think for sbt this means an `=` relationship, i.e. a newer version of the dep does not satisfy the requirement. It also makes sense that major releases should happen together.

My main reason for asking about this is that sometimes different libraries could want different deps, for example my radio baseband might want a newer version of chisel than rocket, but I want to use them together. Because of the way binary compatibility works with chisel stuff, it would be hard to do the normal scala thing and have a `>=` dependency constraint. It's probably better to explicitly say you are not using a standard release (which may come with a recommendation that you re-run tests) if you are bumping deps.

Perhaps it should have subsections that say something like:

* A new minor release of rocket may update firrtl, chisel, and rocket-tools to a newer minor release of those dependencies. 
* A new major release of rocket may update firrtl, chisel, and rocket tools to a newer major release of those dependencies.

Reply:
@grebe take a look at https://github.com/chipsalliance/rocket-chip/pull/2284/files#diff-baa362176b63e6912fc6721235a68502R41 and let me know if this is more clear to you